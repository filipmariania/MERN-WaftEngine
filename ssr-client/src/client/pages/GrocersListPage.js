import React from 'react';
import { Helmet } from 'react-helmet';
import { connect } from 'react-redux';
import MapComponent from '../components/MapComponent';
import SearchBox from '../components/SearchBox';
import fetchGrocers from '../actions/fetchGrocers';
import GrocersTable from '../components/GrocersTable';

export class GrocersListPage extends React.Component {
  state = { isShown: false };
  componentDidMount() {
    this.handleSearch();
    this.setState({ isShown: true });
  }
  handleSearch = () => {
    this.props.fetchGrocers();
  };
  head = () => {
    return (
      <Helmet>
        <title>{`${this.props.grocers.length} Grocers Loaded`}</title>
        <meta property="og:title" content="Grocers App" />
      </Helmet>
    );
  };
  render() {
    return (
      <div className="container-fluid" style={{ padding: '20px 0' }}>
        {this.head()}
        <main className="row no-margin-top">
          <div className="col s12 l12">
            <SearchBox search={this.handleSearch} />
            <MapComponent />
            <GrocersTable />
          </div>
        </main>
      </div>
    );
  }
}

const mapStateToProps = ({ grocers }) => ({ grocers });

const mapDispatchToProps = { fetchGrocers };

function loadData(store) {
  return store.dispatch(fetchGrocers());
}

export default {
  loadData,
  component: connect(
    mapStateToProps,
    mapDispatchToProps,
  )(GrocersListPage),
};
