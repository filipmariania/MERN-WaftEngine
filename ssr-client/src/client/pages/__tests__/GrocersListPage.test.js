import React from 'react';
import { create } from 'react-test-renderer';
import { MapList } from '../GrocersListPage';

test('snapshot', () => {
  const c = create(<MapList grocers={[]} searchParams={{ handleSearch: () => null }} />);
  expect(c.toJSON()).toMatchSnapshot();
});
