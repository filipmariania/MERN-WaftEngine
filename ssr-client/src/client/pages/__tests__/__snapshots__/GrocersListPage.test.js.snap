// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="container-fluid"
  style={
    Object {
      "padding": "20px 0",
    }
  }
>
  <main
    className="row"
  >
    <div
      className="col s12 l6"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="row"
        >
          <div
            className="input-field col s3"
          >
            <label
              htmlFor="latitudeSearch"
            >
              Latitude
              <input
                id="latitudeSearch"
                name="latitude"
                onChange={[Function]}
                placeholder="latitude"
                value=""
              />
              <span
                className="helper-text"
              />
            </label>
          </div>
          <div
            className="input-field col s3"
          >
            <label
              htmlFor="longitudeSearch"
            >
              Longitude
              <input
                id="longitudeSearch"
                name="longitude"
                onChange={[Function]}
                placeholder="longitude"
                value=""
              />
              <span
                className="helper-text"
              />
            </label>
          </div>
          <div
            className="input-field col s3"
          >
            <label
              htmlFor="distanceSearch"
            >
              Distance in Meter
              <input
                id="distanceSearch"
                name="distance"
                onChange={[Function]}
                placeholder="distance"
                value=""
              />
              <span
                className="helper-text"
              />
            </label>
          </div>
          <div
            className="col s3"
          >
            <label
              htmlFor="searchBtn"
            >
               
              <button
                className="btn"
                id="searchBtn"
              >
                Submit
              </button>
            </label>
          </div>
        </div>
      </form>
      <table
        className="responsive-table"
      >
        <thead>
          <tr>
            <th>
              Store Brand
            </th>
            <th>
              Address
            </th>
            <th>
              City
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <div
      className="col s12 l6"
      style={
        Object {
          "overflow": "hidden",
        }
      }
    >
      <div
        style={
          Object {
            "height": "calc(100vh - 125px)",
            "width": "100%",
          }
        }
      >
        <div
          onClick={[Function]}
          onMouseDownCapture={[Function]}
          onMouseMove={[Function]}
          style={
            Object {
              "height": "100%",
              "margin": 0,
              "padding": 0,
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <div
            style={
              Object {
                "height": "100%",
                "left": 0,
                "margin": 0,
                "padding": 0,
                "position": "absolute",
                "top": 0,
                "width": "100%",
              }
            }
          />
          <div
            style={
              Object {
                "height": "50%",
                "left": "50%",
                "margin": 0,
                "padding": 0,
                "position": "absolute",
                "top": "50%",
                "width": "50%",
              }
            }
          >
            <div
              style={
                Object {
                  "height": "100%",
                  "left": 0,
                  "margin": 0,
                  "padding": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": "100%",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
